### PR TITLE
add initial %%run support to zigbot

### PR DESCRIPTION
**Pull request by g-w1**
Ignore the hzzp changes, just taking it out of a submodule because there is one change I had to make by removing some error handling that caught non errors. PLEASE make sure it works on your machine; but it works on mine.

Do we want to support people doing 
\`\`\`lang
some zig code
\`\`\`
?
If so, I can update the tokenizer?

### CODE REVIEW

**Comment by g-w1** [comment]
We could also do a thing where it fills in 
```zig
const std = @import("std");
pub fn main() {
<user code here>
}
```
if there is no main supplied.

File: src/main.zig [hunk 1]
```diff
@@ -507,10 +608,7 @@ pub fn main() !void {
                             state = if (c == '%') .ready else .no_match;
                         },
                         .ready => {
-                            switch (c) {
-                                ' ', ',', '\n', '\t', '(', ')', '!', '?', '[', ']', '{', '}' => break,
-                                else => try buffer.append(c),
-                            }
```
Comment (fengb) [inline src/main.zig:513]: This breaks existing functionality. We won't be able to embed %%dab in the middle of the sentence.

We could have a separate state _just_ for detecting run. Something like
```zig
.ready => {
    if (std.mem.eql(u8, buffer.slice(), "run")) state = .run;
    switch (c) { ... }
},
.run => try buffer.append(c),
```

File: src/main.zig [hunk 2]
```diff
@@ -250,6 +250,15 @@ const Context = struct {
             }),
             else => {},
         }
+        if (std.mem.startsWith(u8, ask, "run")) {
+            const ran = try self.parseRunAndRun(ask);
+
+            return try self.sendDiscordMessage(.{
+                .channel_id = channel_id,
+                .title = "Run Results:",
+                .description = try std.fmt.allocPrint(self.allocator, "**stdout**: {s}\n**stderr**: {s}", .{ ran.stdout.slice(), ran.stderr.slice() }),
```
Comment (fengb) [inline src/main.zig:259]: This is currently leaking. I would prefer a non-allocating, but I'm not quite sure how to arrange the code.

File: src/main.zig [hunk 3]
```diff
@@ -363,7 +410,61 @@ const Context = struct {
...
+                },
+                else => unreachable,
+            }
+        }
+        return result;
+    }
+    const GithubIssue = struct { repo: Buffer(0x100), number: u32, title: Buffer(0x1000), url: Buffer(0x100) };
+    const RunResult = struct {
+        stdout: Buffer(0x1000) = .{},
+        stderr: Buffer(0x1000) = .{},
```
Comment (fengb) [inline src/main.zig:466]: Buffer() is passed around on the stack. At this size, I would prefer if this was allocating instead.

Comment (g-w1) [reply src/main.zig:466]: This would make a lot more allocations, since I had to change buffer to 0x1000 in a lot of places. Do we want this?

**Pull request merged**